projects/media-viewer/src/lib/media-viewer/viewers/image-viewer/image-viewer.component.ts

```diff
@@ -38,15 +40,20 @@ export class ImageViewerComponent {
   @Input()
   set printOperation(operation: PrintOperation | null) {
     if (operation) {
-      const printWindow = window.open(this.url);
-      printWindow.print();
+      this.printService.printDocumentNatively(this.url);
     }
   }
 
   @Input()
   set downloadOperation(operation: DownloadOperation | null) {
     if (operation) {
-      // TODO
+      const a = document.createElement('a');
+      document.body.appendChild(a);
+      a.setAttribute('style', 'display: none');
+      a.href = this.url;
+      a.download = this.downloadFileName;
+      a.click();
+      a.remove();
```

Comment:
I'm assuming there is no better way to do this?

Reply:
it seems like the most reliable solution...